### PR TITLE
fix: `init.f90` file generates ast and asr with cc flag

### PR DESCRIPTION
**Pull request by parth121101**
Towards: #5463 , #5781

### CODE REVIEW

**Comment by parth121101** [comment]
Following is `init.f90` fortran file 

```
cat tests/errors/init1.f90
program init1
integer x = 1
end
```

Current main branch is unable to generate ast and asr with cc flag 
```
lfortran tests/errors/init1.f90 --continue-compilation --show-asr
syntax error: Invalid syntax for variable initialization (try inserting '::' after the type)
 --> tests/errors/init1.f90:2:1
  |
2 | integer x = 1
  | ^^^^^^^^^^^^^ 


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.
```

While with this PR we are able to generate ast as well as ast.
```
lfortran tests/errors/init1.f90 --continue-compilation --show-asr
syntax error: Invalid syntax for variable initialization (try inserting '::' after the type)
 --> tests/errors/init1.f90:2:1
  |
2 | integer x = 1
  | ^^^^^^^^^^^^^ 


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.
(TranslationUnit
    (SymbolTable
        1
        {
            init1:
                (Program
                    (SymbolTable
                        2
                        {
                            x:
                                (Variable
                                    2
                                    x
                                    []
                                    Local
                                    (IntegerConstant 1 (Integer 4) Decimal)
                                    (IntegerConstant 1 (Integer 4) Decimal)
                                    Save
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.
                                )
                        })
                    init1
                    []
                    []
                )
        })
    []
)
```

File: tests/reference/asr-continue_compilation_2-a6145a1.stderr [hunk 1]
```diff
@@ -98,490 +125,481 @@ semantic error: Unexpected args, SetExponent expects (real, int) as arguments
...
 
-semantic error: Cannot assign to a constant variable
-  --> tests/errors/continue_compilation_2.f90:66:5
-   |
-66 |     x = 1
-   |     ^^^^^ assignment here
-   |
-38 |     integer, parameter :: x = 2
-   |                           ~~~~~ declared as constant
-
```
Comment (certik) [inline tests/reference/asr-continue_compilation_2-a6145a1.stderr:120]: This error does not happen anymore. This is due to the new "x" shadowing the old "x".

File: tests/errors/continue_compilation_2.f90 [hunk 2]
```diff
@@ -29,6 +29,10 @@ program continue_compilation_2
     use Geometry
     implicit none
 
+    integer, pointer, parameter :: v => null()
+    integer, allocatable, parameter :: v=1
+    integer x = 1
```
Comment (certik) [inline tests/errors/continue_compilation_2.f90:34]: Rename `x` to some unique name, e.g., if `y` is not used:
```suggestion
    integer y = 1
```
In order to fix the above error.

**Review by certik — COMMENTED** [review]
Otherwise I think this looks good!

**Review by certik — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**